### PR TITLE
fix: DH-22738: Fix capacity edge-case when encoding binary arrays

### PR DESCRIPTION
**Pull request by robbcamera**

### CODE REVIEW

File: extensions/parquet/table/src/test/java/io/deephaven/parquet/table/TestParquetTools.java [hunk 1]
```diff
@@ -212,6 +212,27 @@ public void testWriteTable() {
...
+                .setCompressionCodecName("UNCOMPRESSED")
+                .useDictionary("StrValue", false)
+                .setMaximumDictionaryKeys(0)
+                .build();
+
+        String path = testRoot + File.separator + "emptyStringsTable.parquet";
+        ParquetTools.writeTable(emptyStringsTable, path, instructions);
+        Table result = ParquetTools.readTable(path);
+        TableTools.show(result);
+        TableTools.show(emptyStringsTable);
```
Comment (devinrsmith) [inline extensions/parquet/table/src/test/java/io/deephaven/parquet/table/TestParquetTools.java]: remove

File: extensions/parquet/table/src/main/java/io/deephaven/parquet/table/ParquetTableWriter.java [hunk 2]
```diff
@@ -263,7 +263,7 @@ private static void write(
                 try {
                     writeColumnSource(tableRowSet, writeInstructions, rowGroupWriter, computedCache, columnName,
                             columnSource);
-                } catch (IllegalAccessException e) {
+                } catch (final RuntimeException | IllegalAccessException e) {
```
Comment (devinrsmith) [inline extensions/parquet/table/src/main/java/io/deephaven/parquet/table/ParquetTableWriter.java]: This is a good change.

I would go further - writeColumnSource declares it throws `IllegalAccessException`, but never does, it should just be removed.

File: extensions/parquet/base/src/main/java/io/deephaven/parquet/base/PlainBinaryChunkedWriter.java [hunk 3]
```diff
@@ -138,7 +138,7 @@ public WriteResult writeBulkFilterNulls(@NotNull final Binary[] bulkValues,
     }
 
     private void ensureCapacityFor(@NotNull final Binary v) {
-        if (v.length() == 0 || innerBuffer.remaining() >= v.length() + Integer.BYTES) {
+        if (innerBuffer.remaining() >= v.length() + Integer.BYTES) {
```
Comment (devinrsmith) [inline extensions/parquet/base/src/main/java/io/deephaven/parquet/base/PlainBinaryChunkedWriter.java]: I would slightly re-do this method to make it more "obvious" - currently, we are requiring the method itself to be aware of the extra 4 bytes for the length. It would be much better to just have the caller pass in the remaining size they need available, and to make the method name a bit "better":

```java
        ensureRemaining(Integer.BYTES + v.length());
        innerBuffer.putInt(v.length());
        innerBuffer.put(v.toByteBuffer())
```

**Review by devinrsmith — CHANGES_REQUESTED** [review]
Good triage. Can you audit other similar code in parquet writing to make sure we aren't making similar ensure capacity mistakes?

**Review by devinrsmith — APPROVED** [review]
